python/test/test_github_service.py

```diff
@@ -99,6 +101,109 @@ def test_sets_up_class(self, mock_github_client_core_api, mock_github_client_gql
+                         repo_name: str = None, has_commits: bool = True) -> Mock:
+        repository_to_consider_for_archiving = Mock(
+            Repository, archived=archived, fork=fork)
+        repository_to_consider_for_archiving.name = repo_name
+        repository_to_consider_for_archiving.get_commits.return_value = [
+            Mock(Commit, author=Mock(NamedUser, date=last_active_date))] if has_commits else None
+        return repository_to_consider_for_archiving
+
+    def setUp(self):
+        self.last_active_cutoff_date = datetime.now()
+        self.before_cutoff = self.last_active_cutoff_date - timedelta(days=1)
+        self.after_cutoff = self.last_active_cutoff_date + timedelta(days=1)
+
+    def test_no_archive_when_repo_is_archived(self, mock_github_client_core_api):
+        repo = self.__get_repository(
+            last_active_date=self.after_cutoff, archived=True)
+        mock_github_client_core_api.return_value.get_organization().get_repos.return_value = [
+            repo,
+            repo,
+            repo
+        ]
+        GithubService("", ORGANISATION_NAME).archive_inactive_repositories(
+            "all", self.last_active_cutoff_date, [])
+        self.assertEqual(repo.edit.called, False)
+
+    def test_no_archive_when_repo_is_forked(self, mock_github_client_core_api):
+        repo = self.__get_repository(
+            last_active_date=self.after_cutoff, fork=True)
+        mock_github_client_core_api.return_value.get_organization().get_repos.return_value = [
+            repo,
+            repo,
+            repo
+        ]
+        GithubService("", ORGANISATION_NAME).archive_inactive_repositories(
+            "all", self.last_active_cutoff_date, [])
+        self.assertEqual(repo.edit.called, False)
+
+    def test_no_archive_when_recently_active(self, mock_github_client_core_api):
+        repo = self.__get_repository(last_active_date=self.after_cutoff)
+        mock_github_client_core_api.return_value.get_organization().get_repos.return_value = [
+            repo,
+            repo,
+            repo
+        ]
+        GithubService("", ORGANISATION_NAME).archive_inactive_repositories(
+            "all", self.last_active_cutoff_date, [])
+        self.assertEqual(repo.edit.called, False)
+
+    def test_no_archive_when_on_allow_list(self, mock_github_client_core_api):
+        repo = self.__get_repository(
+            last_active_date=self.after_cutoff, repo_name="allow_me")
+        mock_github_client_core_api.return_value.get_organization().get_repos.return_value = [
+            repo,
+            repo,
+            repo
+        ]
+        GithubService("", ORGANISATION_NAME).archive_inactive_repositories("all", self.last_active_cutoff_date,
+                                                                           ["allow_me"])
+        self.assertEqual(repo.edit.called, False)
+
+    def test_no_archive_when_repo_has_no_commits_even_if_before_cutoff(self, mock_github_client_core_api):
```

Comment:
This test highlights something I considered a bit odd, the logic for the archiving seemed to allow repos that have no commits to live forever ([REF](https://github.com/ministryofjustice/operations-engineering/blob/abe29b422539fbdde4dca1a9fb50131d99101648/python/scripts/archive_repositories.py#LL48C7-L48C7))

I've copied the logic here as I didn't want to refactor any business rules in this PR, but thought I'd point it out for thoughts 🤔

IMO a repo that has had no commits and is over 1 year and a half old, should be archived - but maybe there's some information in the grapevine that I'm not aware of 🍇

Reply:
If you can put in some logic that gets the date of the repo created and no commits in last 18 months then go ahead, I think the code looks at the last time the main branch was updated, before that it was checking when the repo was last updated which was skewed by the dependabot PR's which stopped the repo's by archived

Reply:
It looks like you can get a created [created_at](https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28#get-a-repository) property from the repositories👀

Will help with the exception case as well! But I'll leave that for a separate PR 😁